Sources/AsyncHTTPClient/ConnectionPool/HTTPConnectionPool+HTTP1State.swift

```diff
@@ -0,0 +1,694 @@
+                        // eventLoop (check in guard), the request has an eventLoopRequirement.
+                        let eventLoop = nextWaiter.eventLoopRequirement!
+                        let newConnection = HTTP1ConnectionState(
+                            connectionID: self.idGenerator.next(),
+                            eventLoop: eventLoop,
+                            waiter: nextWaiter
+                        )
+                        self.connections[index] = newConnection
+                        return .init(
+                            .none,
+                            .replaceConnection(connectionState.close(), with: newConnection.connectionID, on: eventLoop)
+                        )
+                    }
+
+                    let connection = connectionState.lease()
+                    self.connections[index] = connectionState
+                    return .init(
+                        .executeRequest(nextWaiter.request, connection, cancelWaiter: nextWaiter.requestID),
+                        .none
+                    )
+                }
+
+                self.connections[index] = connectionState
+                return .init(.none, .scheduleTimeoutTimer(connectionState.connectionID))
+
+            case .shuttingDown(unclean: let unclean):
+                // if we are in shutdown, we want to get rid off this connection asap.
+                guard connectionState.started(connection) == nil else {
+                    preconditionFailure("Expected to remove the waiter when shutdown is issued")
+                }
+
+                self.connections.remove(at: index)
+                let isShutdown: StateMachine.ConnectionAction.IsShutdown
+                if self.connections.isEmpty {
+                    self.state = .shutDown
+                    isShutdown = .yes(unclean: unclean)
+                } else {
+                    isShutdown = .no
+                }
+
+                return .init(.none, .closeConnection(connectionState.close(), isShutdown: isShutdown))
+
+            case .shutDown:
+                preconditionFailure("The pool is already shutdown all connections must already been torn down")
+            }
+        }
+
+        mutating func failedToCreateNewConnection(_ error: Error, connectionID: Connection.ID) -> Action {
+            guard let index = self.connections.firstIndex(where: { $0.connectionID == connectionID }) else {
+                preconditionFailure("We tried to create a new connection, that we know nothing about?")
+            }
+
+            var connectionState = self.connections[index]
+
+            switch self.state {
+            case .running:
+                var requestAction: StateMachine.RequestAction = .none
+                if let failedWaiter = connectionState.failed() {
+                    requestAction = .failRequest(failedWaiter.request, error, cancelWaiter: failedWaiter.requestID)
```

Comment:
This should not be necessary: if we failed to connect, we should try again (on an exponential backoff) until we either succeed or have more connection attempts in flight than waiters.